src/main/java/bio/terra/workspace/service/status/BaseStatusService.java

```diff
@@ -1,23 +1,25 @@
-package bio.terra.workspace.common.utils;
+package bio.terra.workspace.service.status;
 
+import bio.terra.workspace.common.utils.StatusSubsystem;
 import bio.terra.workspace.generated.model.SystemStatus;
 import bio.terra.workspace.generated.model.SystemStatusSystems;
 import java.util.Date;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicBoolean;
 import org.springframework.scheduling.annotation.Scheduled;
 
-/*
- BaseStatusService is a Spring replacement for workbench-libs' HealthMonitor utilities. It checks
- status information from subsystems asynchronously at regular intervals and provides a cached
- version of the latest statuses to support a high-traffic status endpoint.
- It also tracks time since the last update and returns an unhealthy status if subsystems are not
- checked after some amount of time, which indicates that something has gone wrong.
-
- Specific services should extend this class with Component objects that register the appropriate
- subsystems.
-*/
-public class BaseStatusService {
+/**
+ * BaseStatusService is a Spring replacement for workbench-libs' HealthMonitor utilities.
+ *
+ * <p>It checks status information from subsystems asynchronously at regular intervals and provides
+ * a cached version of the latest statuses to support a high-traffic status endpoint. It also tracks
+ * time since the last update and returns an unhealthy status if subsystems are not checked after
+ * some amount of time, which indicates that something has gone wrong.
+ *
+ * <p>Specific services should extend this class with Component objects that register the
+ * appropriate subsystems.
```

Comment:
I don't really understand why specific services need to extend the class (is-a) instead of having a member variable BaseStatusService (has-a). But if we're saying this needs to be extended, should it be abstract?

Reply:
My thinking on this was that each service has a specific implementation of the BaseStatusService - each of those implementations is a BaseStatusService with a particular set of dependencies. A big driver for this pattern was separating the parts common to other services (which would live in a common library somewhere if we had one) from the actual WM dependencies. When this was written TDR was planning to use it as well, I don't know if that ever happened or got deprioritized.

Yeah, it should probably be abstract.